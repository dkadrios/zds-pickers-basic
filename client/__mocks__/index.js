export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
