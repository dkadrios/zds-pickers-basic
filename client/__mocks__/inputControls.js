export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
