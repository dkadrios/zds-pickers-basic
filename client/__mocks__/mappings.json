{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
