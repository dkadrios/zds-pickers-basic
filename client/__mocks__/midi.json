{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
