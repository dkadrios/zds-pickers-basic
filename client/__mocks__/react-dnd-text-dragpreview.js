/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
export const createDragPreview = () => <img src={null} />
