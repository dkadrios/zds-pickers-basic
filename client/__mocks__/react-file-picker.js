/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
