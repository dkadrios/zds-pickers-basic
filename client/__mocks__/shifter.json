{
  "accessGranted": true,
  "errorMessage": "",
  "errorVisible": false,
  "exportBuffer": [],
  "exportDialogVisible": false,
  "exportFilename": "zds-shifter-backup.txt",
  "found": true,
  "hardwareTestVisible": true,
  "importDialogVisible": false,
  "importInProcess": false,
  "invalidSettingsFile": null,
  "midiActivityLEDMode": 0,
  "midiInActivity": false,
  "midiOutActivity": false,
  "ready": true,
  "resetInProcess": false,
  "responding": true,
  "searchedForShifter": true,
  "serialMidiOutEnabled": true,
  "showResetDialog": false,
  "testInterfaceFound": false,
  "usbMidiOutEnabled": true
}
