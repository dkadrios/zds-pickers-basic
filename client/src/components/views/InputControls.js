import React from 'react'
import { useSelector } from 'react-redux'
import Grid from '@material-ui/core/Grid'
import makeStyles from '@material-ui/core/styles/makeStyles'
import { margin, padding } from 'polished'
import InputControl from '../controls/InputControl'
import MidiActivity from '../MidiActivity'
import LEDModePicker from '../controls/LEDModePicker'
import MidiMenu from '../MidiMenu'
import WaitingOnShifter from '../WaitingOnShifter'
import { stateInputControls, stateShifter } from '../../selectors'

const width = 680

const useStyles = makeStyles(({ constants, palette }) => ({
  root: {
    height: constants.inputControlsHeight,
    ...margin(0, 50),
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    color: palette.text.inverted,

    '& > div': {
      minWidth: width,
      maxWidth: width,
      minHeight: 250,
      display: 'flex',
      flexFlow: 'column nowrap',
      alignItems: 'flex-start',
      borderRadius: 15,
      ...padding(0, 10),
      backgroundColor: palette.accentLight,
      // @include dropShadow($color-shadow, 3px);

      '& > section': {
        display: 'flex',
        flexFlow: 'row nowrap',
        justifyContent: 'space-between',
        alignItems: 'center',
        minWidth: width,
        minHeight: 200,
      },
    },
  },
  midiLedCont: {
    display: 'flex',
    flexFlow: 'row nowrap',
    ...margin(8, 0, 0, 10),
    ...padding(15, 2, 0, 8),
    borderRadius: 6,
    backgroundColor: palette.accent,
    // boxShadow: 'inset 1px 1px 0 $color-shifter-red-accent',
    maxHeight: 29,

    '& svg': {
      marginTop: 5,
    },

    '& button': {
      margin: 0,
      padding: 0,
      position: 'relative',
      top: -6,
      color: palette.text.inverted,
      fontSize: 14,
      '& svg': {
        fill: palette.text.inverted,
      },
    },
  },
}), { name: 'InputControls' })

export const InputControls = () => {
  const inputControls = useSelector(stateInputControls)
  const {
    found,
    midiActivityLEDMode,
    ready,
    responding,
    serialMidiOutEnabled,
    usbMidiOutEnabled,
  } = useSelector(stateShifter)

  const classes = useStyles()

  const visibleControls = inputControls.filter((control, idx) => idx < 3)

  return (
    <div className={classes.root}>
      <div>
        {ready ? (
          <>
            <Grid
              alignItems="center"
              container
              justify="space-between"
            >
              <Grid item>
                {Boolean(visibleControls.length > 0 && ready) && (
                  <div className={classes.midiLedCont}>
                    <LEDModePicker
                      selectedValue={midiActivityLEDMode}
                      serialMidiOutEnabled={serialMidiOutEnabled}
                      usbMidiOutEnabled={usbMidiOutEnabled}
                    >
                      <MidiActivity label="PWR/Midi" />
                    </LEDModePicker>
                  </div>
                )}
              </Grid>
              <Grid item>
                {Boolean(visibleControls.length > 0 && ready) && (
                  <div
                    className={classes.midiLedCont}
                    style={{ marginRight: 10 }}
                  >
                    <MidiMenu />
                  </div>
                )}
              </Grid>
            </Grid>
            <section>
              {visibleControls.map((control, idx) => (
                <InputControl
                  key={idx}
                  {...control}
                  layout={idx === 1 ? 'right' : 'left'}
                />
              ))}
            </section>
          </>
        ) : (
          <section>{Boolean(found && responding) && <WaitingOnShifter />}</section>
        )}
      </div>
    </div>
  )
}

export default InputControls
