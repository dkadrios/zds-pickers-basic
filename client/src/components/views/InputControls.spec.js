import { inputControls, shifter, store, version } from '../../../__mocks__'
import { InputControls, mapDispatchToProps, mapStateToProps } from './InputControls'

const setFlags = jest.fn()

const props = {
  inputControls,
  version,
  shifter,
  setFlags,
}

describe('InputControls tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(InputControls)(props).toMatchSnapshot()
  })

  it('should utilize mapStateToProps', () => {
    expect(mapStateToProps(store)).toEqual({ inputControls, shifter, version })
  })

  it('should utilize mapDispatchToProps', () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).setFlags(3, true, false)
    expect(dispatch.mock.calls[0][0]).toEqual({
      midiActivityLEDMode: 3,
      serialMidiOutEnabled: true,
      type: 'SET_FLAGS',
      usbMidiOutEnabled: false,
    })
  })
})
