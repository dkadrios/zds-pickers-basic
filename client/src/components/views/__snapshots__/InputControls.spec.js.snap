// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControls tests renders correctly 1`] = `
<div>
  <div>
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="space-between"
    >
      <WithStyles(Grid)
        item={true}
      >
        <div>
          <LEDModePicker
            selectedValue={0}
            serialMidiOutEnabled={true}
            setFlags={[MockFunction]}
            usbMidiOutEnabled={true}
          >
            <Connect(MidiActivity)
              label="PWR/Midi"
            />
          </LEDModePicker>
        </div>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
      >
        <div
          style={
            Object {
              "marginRight": 10,
            }
          }
        >
          <Connect(MidiMenu) />
        </div>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <section>
      <InputControl
        active={true}
        calibrationHigh={127}
        calibrationLow={0}
        channel={10}
        controlId={0}
        curve={0}
        inputControls={
          Array [
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 0,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 11,
              "value": 110,
            },
            Object {
              "active": false,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 1,
              "curve": 0,
              "latching": 1,
              "lit": false,
              "polarity": 0,
              "status": 11,
              "value": 111,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 2,
              "curve": 0,
              "latching": 0,
              "lit": false,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 3,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 4,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 5,
              "curve": 0,
              "latching": 0,
              "lit": false,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
          ]
        }
        key="0"
        latching={0}
        layout="left"
        lit={true}
        polarity={0}
        setFlags={[MockFunction]}
        shifter={
          Object {
            "accessGranted": true,
            "errorMessage": "",
            "errorVisible": false,
            "exportBuffer": Array [],
            "exportDialogVisible": false,
            "exportFilename": "zds-shifter-backup.txt",
            "found": true,
            "hardwareTestVisible": true,
            "importDialogVisible": false,
            "importInProcess": false,
            "invalidSettingsFile": null,
            "midiActivityLEDMode": 0,
            "midiInActivity": false,
            "midiOutActivity": false,
            "ready": true,
            "resetInProcess": false,
            "responding": true,
            "searchedForShifter": true,
            "serialMidiOutEnabled": true,
            "showResetDialog": false,
            "testInterfaceFound": false,
            "usbMidiOutEnabled": true,
          }
        }
        status={11}
        value={110}
        version={
          Object {
            "checked": false,
            "checking": false,
            "client": 2.5,
            "firmware": 1.5,
          }
        }
      />
      <InputControl
        active={false}
        calibrationHigh={127}
        calibrationLow={0}
        channel={10}
        controlId={1}
        curve={0}
        inputControls={
          Array [
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 0,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 11,
              "value": 110,
            },
            Object {
              "active": false,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 1,
              "curve": 0,
              "latching": 1,
              "lit": false,
              "polarity": 0,
              "status": 11,
              "value": 111,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 2,
              "curve": 0,
              "latching": 0,
              "lit": false,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 3,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 4,
              "curve": 0,
              "latching": 0,
              "lit": true,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
            Object {
              "active": true,
              "calibrationHigh": 127,
              "calibrationLow": 0,
              "channel": 10,
              "controlId": 5,
              "curve": 0,
              "latching": 0,
              "lit": false,
              "polarity": 0,
              "status": 9,
              "value": 110,
            },
          ]
        }
        key="1"
        latching={1}
        layout="right"
        lit={false}
        polarity={0}
        setFlags={[MockFunction]}
        shifter={
          Object {
            "accessGranted": true,
            "errorMessage": "",
            "errorVisible": false,
            "exportBuffer": Array [],
            "exportDialogVisible": false,
            "exportFilename": "zds-shifter-backup.txt",
            "found": true,
            "hardwareTestVisible": true,
            "importDialogVisible": false,
            "importInProcess": false,
            "invalidSettingsFile": null,
            "midiActivityLEDMode": 0,
            "midiInActivity": false,
            "midiOutActivity": false,
            "ready": true,
            "resetInProcess": false,
            "responding": true,
            "searchedForShifter": true,
            "serialMidiOutEnabled": true,
            "showResetDialog": false,
            "testInterfaceFound": false,
            "usbMidiOutEnabled": true,
          }
        }
        status={11}
        value={111}
        version={
          Object {
            "checked": false,
            "checking": false,
            "client": 2.5,
            "firmware": 1.5,
          }
        }
      />
    </section>
  </div>
</div>
`;
